torchvision/io/_video_opt.py

```diff
@@ -470,7 +484,9 @@ def _probe_video_from_memory(video_data):
     return info
 
 
-def _read_video(filename, start_pts=0, end_pts=None, pts_unit="pts"):
+def _read_video(
+    filename: str, start_pts: int = 0, end_pts: Optional[float] = None, pts_unit: str = "pts"
```

Comment:
we are in a complicated situation here because the type of `start_pts` and `end_pts` depend on `pts_unit`.

They are both floats if `pts_unit == 'sec'`, and both int if `pts_unit == 'pts'`. But I would rather not make this a `Union` here because it will hinder support for torchscript, specially in this file which is currently used in some production systems (with tests not necessarily showing up in GitHub's CI).

Reply:
We do not need a `Union` here, since `int` is a subtype of `float` for `mypy`

```python
foo: int = 1.0
bar: float = 1
```

```
error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
```

This is only a problem if we use `start_pts` or `end_pts` as indexes. But then, I need to take another look why we are trying to do float indexing in the first place.

Reply:
We don't do float indexing. We just use `inf` as a way of saying: go until the end of the file. We could have used a different representation, like `-1` for this, but it would have made a few things a bit more complicated I think